### PR TITLE
Changes needed for future rstan

### PR DESCRIPTION
**Pull request by andrjohns**
Hi,

This PR adds the changes to compiler and linker flags needed for the upcoming version of rstan/StanHeaders. All that was needed in this case was adding the RcppParallel include files and linker flags.

Let me know if you need any more info.

Thanks!
Andrew

### CODE REVIEW

**Comment by mattfidler** [comment]
Thanks @andrjohns 

Running other checks.  It is likely just fine

**Comment by mattfidler** [comment]
Somehow this fails.

**Comment by mattfidler** [comment]
I am re-running the base to see if it still works

File: inst/tools/workaround.R [hunk 1]
```diff
@@ -26,7 +27,7 @@ for (f in c("inst/include/RxODE_RcppExports.h", "src/RcppExports.cpp")) {
                                .badStan)),
             .in)
 
-.in <- gsub("@SL@", "", ##paste(capture.output(StanHeaders:::LdFlags()), capture.output(RcppParallel:::RcppParallelLibs())),
```
Comment (mattfidler) [inline inst/tools/workaround.R:29]: @andrjohns 

Maybe try commenting this out again.  The main branch seems to be working OK.

**Review by mattfidler — CHANGES_REQUESTED** [review]

**Comment by andrjohns** [comment]
Sorry about that, I updated the call to `StanHeaders:::LdFlags()` to satisfy Codefactor but it caused a different error. I've submitted the fix now

**Comment by andrjohns** [comment]
I've re-run these changes locally on Windows and I can't replicate the R CMD check error from the windows testing above, would you be able to test locally as well?

**Comment by andrjohns** [comment]
Apologies that this has been such a process. I was able to replicate the failure, it was caused by Windows linking against the wrong architecture for `RcppParallel` (i.e., was trying to link to 64-bit during 32-bit compilation)

**Pull request closed without merging**